cassandra/src/main/java/se/tre/freki/storage/cassandra/CassandraStore.java

```diff
@@ -371,19 +379,117 @@ public LabelId apply(final ResultSet rows) {
 
   @Nonnull
   @Override
-  public ListenableFuture<LabelMeta> getMeta(final LabelId uid,
-                                             final LabelType type) {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ListenableFuture<Optional<LabelMeta>> getMeta(final LabelId id,
+                                                       final LabelType type) {
+
+    final ListenableFuture<Optional<Row>> metas = getOptionalMeta(id, type);
+    return transform(metas, new Function<Optional<Row>, Optional<LabelMeta>>() {
+      @Nullable
+      @Override
+      public Optional<LabelMeta> apply(final Optional<Row> result) {
+        if (!result.isPresent()) {
+          return Optional.absent();
+        }
+
+        final Row row = result.get();
+
+        return Optional.of(
+            LabelMeta.create(id, type, row.getString("name"), row.getString("meta"), 0));
```

Comment:
Can you use the information stored in the row for all parameters? That might be slightly more safe in case there ever is a big with any CQL statement? Also the creation time should be set to the creation time and not 0.